controllers/disruption_schedule_controller.go

```diff
@@ -0,0 +1,221 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DisruptionScheduleReconciler struct {
+	Client  client.Client
+	Scheme  *runtime.Scheme
+	BaseLog *zap.SugaredLogger
+	log     *zap.SugaredLogger
+}
+
+const (
+	ScheduledAtAnnotation       = chaosv1beta1.GroupName + "/scheduled-at"
+	DisruptionScheduleNameLabel = chaosv1beta1.GroupName + "/disruption-schedule-name"
+)
+
+func (r *DisruptionScheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	r.log = r.BaseLog.With("scheduleNamespace", req.Namespace, "scheduleName", req.Name)
+	r.log.Info("Reconciling DisruptionSchedule")
+
+	instance := &chaosv1beta1.DisruptionSchedule{}
+
+	// retrieve schedule
+	if err := r.Client.Get(context.Background(), req.NamespacedName, instance); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !instance.DeletionTimestamp.IsZero() {
+		// NOTE: add a finalizer if anything needs to be written here
+		// The DisruptionSchedule is being deleted.
+		return ctrl.Result{}, nil
+	}
+
+	disruptions, err := r.getChildDisruptions(ctx, instance)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.updateLastScheduleTime(ctx, instance, disruptions); err != nil {
+		r.log.Errorw("unable to update LastScheduleTime of DisruptionSchedule status", "DisruptionSchedule", instance.Name, "err", err)
```

Comment:
```suggestion
		r.log.Errorw("unable to update LastScheduleTime of DisruptionSchedule status", "err", err)
```

`scheduleName: instance.Name` has already been added to every log, after the `r.log = r.BaseLog.With("scheduleNamespace", req.Namespace, "scheduleName", req.Name)` call on L38